warp10/src/main/java/io/warp10/script/processing/typography/PcreateFont.java

```diff
@@ -45,21 +80,62 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
 
     PFont font = null;
     
-    if (3 == params.size()) {
-      font = pg.parent.createFont(params.get(1).toString(), ((Number) params.get(2)).floatValue());
-    } else if (4 == params.size()) {
-      font = pg.parent.createFont(
-          params.get(1).toString(),
-          ((Number) params.get(2)).floatValue(),
-          Boolean.TRUE.equals(params.get(3)));
-    } else if (5 == params.size()) {
-      font = pg.parent.createFont(
-          params.get(1).toString(),
-          ((Number) params.get(2)).floatValue(),
-          Boolean.TRUE.equals(params.get(3)),
-          params.get(4).toString().toCharArray());
-    }
+    //
+    // Handle the case of remote font files differently, passing the URL
+    // to the defined validator
+    //
+    
+    if (params.get(1).toString().startsWith("http://") || params.get(1).toString().startsWith("https://")) {
+      String url = resolver.resolve(params.get(1).toString());
 
+      InputStream in = null;
+      
+      try {
+        //
+        // FIXME(hbs): support WOFF/WOFF2 on top of TTF/OTF
+        //
+        in = new URL(url).openStream();
```

Comment:
Fails if `url` contains a space. This can be avoided by double escaping in WarpScript ie `%2520`.
Escape in code or to be documented.